torch/csrc/autograd/variable.h

```diff
@@ -384,13 +388,26 @@ struct TORCH_API DifferentiableViewMeta : public AutogradMeta {
   /// version_counter.current_version().
   uint32_t attr_version;
 
+  /// By default we use as_strided to recover views which is more efficient.
+  /// view_fn is only saved when as_strided is not supported.
+  /// If view_fn has value, we use it to recover views in backward.
+  c10::optional<std::function<at::Tensor(const at::Tensor&)>> view_fn_;
+
   CreationMeta creation_meta;
 
   bool requires_grad() const override {
     return requires_grad_ || grad_fn_ || (is_view_ && base_.requires_grad());
   }
 
-  DifferentiableViewMeta(at::TensorImpl* self_impl, Variable base,
+  bool support_as_strided() const {
```

Comment:
I am not 100% convinced by this (see my comment above for mixed views with types that support strides and some that don't). Maybe simply call this `.has_view_fn()`? Or delegate this to the base_? It feels like supporting stride is a property of a Tensor, not really of the view meta.